### PR TITLE
164 Update to Drupal 9! 🔥 

### PR DESCRIPTION
**Pull request by leonk**
### Context

> Does this issue have a Trello card?

https://trello.com/c/BR8rzQmp/164-spike-drupal-9-upgrade-impact-assessment

### Intent

This is a follow up PR to https://github.com/ministryofjustice/prisoner-content-hub-backend/pull/211 (~which needs to be deployed to prod first~ which has now been deployed).

This PR updates Drupal to version 9.1.10, and necessary dependencies to their latest versions.

This was actually a relatively straightforward update, helped by the fact we do not have a huge amount of custom code, and that most Drupal modules have already been updated to support Drupal 9.

### Considerations

> Is there any additional information that would help when reviewing this PR?

The previous PR needs to be deployed to production first https://trello.com/c/BR8rzQmp/164-spike-drupal-9-upgrade-impact-assessment
This uninstalls a module that is not compatible with Drupal 9.

### Checklist

- [ ] This PR contains **only** changes related to the above card
- [ ] Tests have been added/updated to cover the change
- [ ] Documentation has been updated where appropriate
- [ ] Tested in Development

### CODE REVIEW

File: docroot/modules/custom/moj_resources/src/TermApiClass.php [hunk 1]
```diff
@@ -70,7 +70,7 @@ private function decorateResponse($term)
     $result['description'] = $term->description[0];
     $result['summary'] = $term->field_content_summary ? $term->field_content_summary->value : '';
     $result['image'] = $term->field_featured_image[0];
-    $result['video'] = $term->field_featured_video[0];
+    $result['video'] = ''; // Field removed, return empty string.
```
Comment (leonk) [inline docroot/modules/custom/moj_resources/src/TermApiClass.php:73]: Same as above

File: composer.json [hunk 2]
```diff
@@ -8,30 +8,26 @@
...
-        "drupal/flysystem": "~1.0",
+        "drupal/flysystem": "^2.0",
         "drupal/flysystem_s3": "^2.0-rc1",
-        "drupal/govuk_design_system": "^1.0@beta",
-        "drupal/govuk_inline_form_errors": "^1.0@beta",
+        "drupal/govuk_design_system": "^2.0",
+        "drupal/govuk_inline_form_errors": "2.0.0-beta2",
         "drupal/health_check": "^1.1",
-        "drupal/jsonapi_cross_bundles": "^1.0@alpha",
+        "drupal/jsonapi_cross_bundles": "dev-3172761-unable-to-install#c0c90e6ecdd894732a0ca58f6d6c314e9dc29f28",
```
Comment (leonk) [inline composer.json:26]: The support for Drupal 9 for this module has not been fully released, so we have to attach to a commit.

File: composer.json [hunk 3]
```diff
@@ -41,10 +37,8 @@
         "drupal/search_api": "^1.17",
         "drupal/select2": "^1.13",
         "drupal/taxonomy_machine_name": "1.x-dev",
-        "drupal/theme_switcher": "^1.1",
+        "drupal/theme_switcher": "dev-8.x-1.x#06bf0a03a8cd6932d4ee8369973128c4915e9eef",
```
Comment (leonk) [inline composer.json:40]: Same as above

File: composer.json [hunk 4]
```diff
@@ -140,12 +136,21 @@
...
             "type": "composer",
             "url": "https://asset-packagist.org"
+        },
+        {
+            "type": "vcs",
+            "url": "https://git.drupalcode.org/issue/jsonapi_cross_bundles-3172761"
+        },
+        {
+            "type": "vcs",
+            "url": "https://git.drupalcode.org/issue/theme_switcher-3154752"
```
Comment (leonk) [inline composer.json:146]: It's not ideal to have to include these repos, but both the changes should eventually be merged into the main branch and released.  So we can remove these after that has happened.

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**